feat: unify weaverbird pipeline failure exceptions (TCTC-9178) (#2238)
ToucanToco/weaverbird: server/CHANGELOG.md

## Unreleased

* Error management improvement: 

>   - Add a global `PipelineFailure` exception to make easier to catch translation and execution errors from supported backends.
>   - Now, on error, Pypika backends raise a `PipelineTranslationFailure` which specify the error, and also the step name and the step index if known. 

Reviewer: ```suggestion
  - A `PipelineFailure` exception was added to make it easier to catch translation and execution errors from supported backends. It is the parent excetpion of both `PipelineTranslationFailure` and `PipelineExecutionFailure`
  - On error, Pypika backends now raise a `PipelineTranslationFailure` which specifies the error, the step name and the step index if known. 
```

Author: [75b5cc8](https://github.com/ToucanToco/weaverbird/pull/2238/commits/75b5cc85f88f163b6c8e32cd51fdaf1f56ff4c56)
